### DISCUSSION TITLE
How contributors communicate

### DISCUSSION THREAD

**Discussion by lorepirri** [tupuio/tupuapp#11] [Ideas]
# Overview

I would like to propose to formalize briefly how the contributors to the Tupu-app communicate.

Feel free to contribute, and submit feedback through the comments below.

The section conclusions will be kept updated by me.

Once this seems to reach stability, will be promoted to a WiKi.

# Conclusions

_tbd_

**Comment by lorepirri** [comment]
**My contribute:** 

I would take into consideration that that volunteer contributors will operate on a non-linear workday, meaning that one can move between work and non-work time on an asynchronous schedule.

I propose to adopt _asynchronous communication_.

GitLab has an awesome and extensive knowledge base regarding [asynchronous communication](https://about.gitlab.com/company/culture/all-remote/asynchronous/). But, it could be disorienting, and maybe unnecessary to adopt all the best [practices stated in their handbook](https://about.gitlab.com/company/culture/all-remote/asynchronous/#best-practices-guidelines-and-async-feature-set).

> An initial team-building sync at the start of a project or milestone can unlock a highly efficient follow-on series of asynchronous events. 

We should be mindful to structure these initial syncs intentionally to build rapport and trust. We could set regular meetings for bootstrapping the app, and onboard new people, and from there go async as much as possible.

> The goal of gathering the group in a shared space should be unambiguous: it serves to equip team members with the context they need to switch primarily to asynchronous afterwards.

A rule of thumb for me, until we better define our workflow, and given that our means of communication are emails, GitHub, and on slack, could be:

- **Slack message**: for quick questions, or volatile messages, anything that it is relevant for the moment and does not need to be relevant for the history and memory of the team. Theoretically, Slack is for instant and sync communication, do not expect an immediate response though. Any important information that should persist, should exists into the handbook or the documentation (the WiKi could serve the handbook, and maybe the documentation).
- **GitHub Discussion**: if I have an idea that I would like to elaborate together with the team, I open a [Discussion on the repo](https://github.com/tupuio/tupuapp/discussions). We could use the WiKi once a topic here in Discussions grows mature.
- **Feature request**: if I have a specific feature to suggest, which is already defined and clear, possibly discussed here or on Discussion, or during a call, I then open a feature request ([issues](https://github.com/tupuio/tupuapp/issues) -> new issue -> [feature request](https://github.com/tupuio/tupuapp/issues/new?assignees=&labels=feature&template=feature_request.md)).  One could create a Draft item in the project board as placeholder, or open a Discussion if the topic deserves a discussion with comments. Generally, in fact, the new features are proposed and discussed before being open, and probably but not necessarily, the discussion is originated or moderated by somebody who is giving the direction of the product (eventually a Product Owner, or Product  Manager).
- **Bug request**: if I notice a bug, I open a bug request ([issues](https://github.com/tupuio/tupuapp/issues) -> new issue -> [bug report](https://github.com/tupuio/tupuapp/issues/new?assignees=&labels=bug&template=bug_report.md)). A bug is something that is not working according to what expected (a crash, or a failure), or specified (according to specifications). If there is some functionality that I would like to add or to be working differently, that is a new feature, not a bug. 
- **Handbook**: is the central repository for how we run the project, who we are and how we communicate. In our case, any mature discussion could be moved to the WiKi, and in there, any other further changes done through PRs (as for any code changes to the codebase).
- **Documentation**: _tbd_, maybe this won't require much more than the WiKi.

In case this make sense, I will further elaborate :) any feedback is appreciated.

**Reply by tsg** [reply]
Thanks for writing this down, @lorepirri, great stuff! My main suggestion would be to keep the process/tools overhead to a minimum and only introduce new tools or processes when the need becomes obvious. I like what you proposed because it keeps almost everything in GitHub, which has simple but usable tools.

I think the general workflow can be:

* A card is added to [the project](https://github.com/orgs/tupuio/projects/1/views/4) with a high level description (e.g. "Mentors should be able to communicate with mentees")
* The card is transformed into an issue with more details. This can be done by the engineer picking up the card or by the person requesting the feature/change (e.g. Madalina). If there are too many unknowns, a GitHub Discussion can be started to figure out the general direction first, or we discuss it in a team sync.
* The engineer that has picked up the card opens a PR to implement it. Generally aim for one PR per feature/bug fix. Ask for reviews.
* While the work is in progress, the card should be in the **In progress** column.
* At least one "Accepted" review is needed to merge. Once the PR is accepted, the original author clicks the "Squash and merge" button.
* We move the card to the **Done** column. Congrats!

I think for the start, the Handbook + Documentation can be a single `CONTRIBUTING.md` file in the tupu-app repo. Or even in the `README.md` file.
